filesystem/iso9660.ksy

```diff
@@ -7,206 +7,84 @@ meta:
+  - susp http://www.ymi.com/ymi/sites/default/files/pdf/Systems%20Use%20P1281.pdf
+  - rrip http://www.ymi.com/ymi/sites/default/files/pdf/Rockridge.pdf
+  - rras http://www.estamos.de/makecd/Rock_Ridge_Amiga_Specific
+  - rrzf https://dev.lovelyhq.com/libburnia/web/wikis/Zisofs
+instances:
+  sector_size:
+    doc-ref: ecma-119 6.1.2
+    value: 0x800
+  volume_descriptor_set:
+    io: _root._io
+    pos: _root.sector_size * 0x10
+    type: volume_descriptor
+    size: sector_size
+    repeat: until
+    repeat-until: _.type == volume_type::volume_descriptor_set_terminator
+enums:
+  volume_type:
+    0x00: boot_record_volume_descriptor
+    0x01: primary_volume_descriptor
+    0x02: supplementary_volume_descriptor
+    0x03: volume_partition_descriptor
+    0xff: volume_descriptor_set_terminator
 types:
-  vol_desc:
-    seq:
-      - id: type
-        type: u1
-      - id: magic
-        contents: "CD001"
-      - id: version
-        type: u1
-      - id: vol_desc_boot_record
-        type: vol_desc_boot_record
-        if: type == 0
-      - id: vol_desc_primary
-        type: vol_desc_primary
-        if: type == 1
-  vol_desc_boot_record:
+  u2bi:
+    doc-ref: ecma-119 7.2.3
     seq:
-      - id: boot_system_id
-        type: str
-        size: 32
-        encoding: UTF-8
-      - id: boot_id
-        type: str
-        size: 32
-        encoding: UTF-8
-  vol_desc_primary:
-    doc-ref: 'http://wiki.osdev.org/ISO_9660#The_Primary_Volume_Descriptor'
+      - id: le
+        type: u2le
+      - id: be
+        type: u2be
+  u4bi:
+    doc-ref: ecma-119 7.3.3
     seq:
-      - id: unused1
-        contents: [0]
-      - id: system_id
-        type: str
-        size: 32
-        encoding: UTF-8
-      - id: volume_id
-        type: str
-        size: 32
-        encoding: UTF-8
-      - id: unused2
-        contents: [0, 0, 0, 0, 0, 0, 0, 0]
-      - id: vol_space_size
-        type: u4bi
-      - id: unused3
-        contents: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
-      - id: vol_set_size
-        type: u2bi
-      - id: vol_seq_num
-        type: u2bi
-      - id: logical_block_size
-        type: u2bi
-      - id: path_table_size
-        type: u4bi
-      - id: lba_path_table_le
-        type: u4le
-      - id: lba_opt_path_table_le
+      - id: le
         type: u4le
-      - id: lba_path_table_be
-        type: u4be
-      - id: lba_opt_path_table_be
+      - id: be
         type: u4be
-      - id: root_dir
-        type: dir_entry
-        size: 34
-      - id: vol_set_id
-        type: str
-        size: 128
-        encoding: UTF-8
-      - id: publisher_id
-        type: str
-        size: 128
-        encoding: UTF-8
-      - id: data_preparer_id
+  datetime_long:
+    doc-ref: ecma-119 8.4.26.1
+    seq:
+      - id: year
         type: str
-        size: 128
-        encoding: UTF-8
-      - id: application_id
+        size: 0x4
```

Comment:
Please don't use hex notation in the `size` key. 

```suggestion
        size: 4
```

Even the `ecma-119` specification uses normally decimal notation, so why do you convert it?

> **8.4.26.1 Date and Time Format**
> 
> The date and time shall be represented by a 17-byte field recorded as follows.
> 
> <table>
> <caption align=center><strong>Table 5 – Date and Time Format</strong></caption>
> <thead>
> <tr>
> <th>RBP</th>
> <th>Interpretation</th>
> <th>Content</th>
> </tr>
> <tbody>
> <tr><td>1 to 4</td><td>Year from I to 9999</td><td>Digits</td></tr>
> <tr><td>5 to 6</td><td>Month of the year from 1 to 12</td><td>Digits</td>
> <tr><td>7 to 8</td><td>Day of the month from 1 to 31</td><td>Digits</td>
> <tr><td>9 to 10</td><td>Hour of the day from 0 to 23</td><td>Digits</td>
> <tr><td>11 to 12</td><td>Minute of the hour from 0 to 59</td><td>Digits</td>
> <tr><td>13 To 14</td><td>Second of the minute from 0 to 59</td><td>Digits</td>
> <tr><td>15 to 16</td><td>Hundredths of a second</td><td>Digits</td>
> <tr><td>17</td><td>Offset from Greenwich Mean Time in number<br>of 15 min intervals from -48 (West) to +52<br> (East) recorded according to 7.1.2</td><td>
> numerical value</td></tr>
> </tbody>
> </table>

I mean, most humans (and even programmers) are still more comfortable with the decimal base than hexadecimal, at least for byte sizes.

Reply:
Removed all size in hex